File: src/main/java/com/github/simiacryptus/aicoder/util/TextBlock.java
package com.github.simiacryptus.aicoder.util;

import java.util.Arrays;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public interface TextBlock {
    CharSequence TAB_REPLACEMENT = "  ";
    String DELIMITER = "\n";

    CharSequence[] rawString();

    default CharSequence getTextBlock() {
        return Arrays.stream(rawString()).collect(Collectors.joining(DELIMITER));
    }

    TextBlock withIndent(CharSequence indent);

    default Stream<CharSequence> stream() {
        return Arrays.stream(rawString());
    }
}

File: src/main/kotlin/com/github/simiacryptus/aicoder/util/TextBlock.kt
package com.github.simiacryptus.aicoder.util

import java.util.*
import java.util.stream.Collectors
import java.util.stream.Stream

interface TextBlock {
    fun rawString(): Array<CharSequence>?
    val textBlock: CharSequence?
        get() = Arrays.stream(rawString()).collect(Collectors.joining(DELIMITER))

    fun withIndent(indent: CharSequence?): TextBlock
    fun stream(): Stream<CharSequence> {
        return Arrays.stream(rawString())
    }

    companion object {
        val TAB_REPLACEMENT: CharSequence = "  "
        const val DELIMITER = "\n"
    }
}